Game.cpp

```diff
@@ -0,0 +1,261 @@
+﻿#include <algorithm>
+
+#include "Game.h"
+#include "Hero.h"
+#include "Map.h"
+
+using namespace std;
+
+unsigned int Game::getMonsterCountInField(const int x, const int y) const
+{
+	return count_if(monsters.begin(), monsters.end(),
+					[x, y](const Monster &monster) { return monster.getX() == x && monster.getY() == y; });
+}
+
+void Game::setMap(const Map &map)
+{
+	if (this->gameState == Game::GameState::started)
+	{
+		throw Game::GameAlreadyStartedException();
+	}
+	else if (hero != nullptr || !monsters.empty())
+	{
+		throw Game::AlreadyHasUnitsException();
+	}
+
+	// Delete map
+	if (this->map != nullptr)
+		delete this->map;
+
+	// Set map
+	this->map = new Map(map);
+}
+
+void Game::putHero(const Hero &hero, const int x, const int y)
+{
+	// Check the availability of the field located at the provided location
+	checkFieldAvailability(x, y);
+
+	if (this->gameState == Game::GameState::started)
+	{
+		throw Game::GameAlreadyStartedException();
+	}
+	else if (this->hero != nullptr)
+	{
+		throw Game::AlreadyHasHeroException();
+	}
+	else
+	{
+		this->hero = new Hero(hero);
```

Comment:
A hero változóbol ki lehet venni a pointerét is

Reply:
Úgy érted ne legyen új Hero, hanem mutasson az átadott referenciára?
Ebben az esetben az eredeti object is változik, ezt nem akartam. (De ez vonatkozik a Monsterekre is)

Reply:
Én úgy vélem, bár meggyőzhető vagyok

Reply:
Végül is egyelőre csak egy Game példány a releváns, így ha nincs erre külön kérés akkor szerintem se legyen másolva.